packages/contract/src/builder.type-test.ts

```diff
@@ -0,0 +1,164 @@
+/**
```

Comment:
perfect, you add type testing. can you integrate those tests in vitest using this guide https://vitest.dev/guide/testing-types.html#concrete-expected-objects-vs-typeargs